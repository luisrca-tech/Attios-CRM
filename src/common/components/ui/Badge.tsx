import { forwardRef } from 'react';
import { cn } from '~/lib/utils';

type BadgeProps = {
	variant?: 'rounded' | 'square' | 'status';
	color?: 'green' | 'white';
	children?: React.ReactNode;
	className?: string;
};

const variants = ({ variant = 'rounded', color = 'green' }: BadgeProps) =>
	cn(
		'flex items-center justify-center px-[0.625rem] py-1 font-bold text-sm leading-5',
		{
			'rounded-xl bg-secondary-200/10 text-secondary-200':
				variant === 'rounded' && color === 'green',
			'rounded-lg bg-secondary-200/10 text-secondary-200':
				variant === 'square' && color === 'green',
			'gap-1 rounded-md bg-white-200 font-normal text-primary-200':
				variant === 'status' && color === 'white'
		}
	);

export const Badge = forwardRef<HTMLDivElement, BadgeProps>(
	({ children, className, ...rest }, ref) => {
		return (
			<div ref={ref} className={cn(variants(rest), className)}>
				{children}
			</div>
		);
	}
);

Badge.displayName = 'Badge';
