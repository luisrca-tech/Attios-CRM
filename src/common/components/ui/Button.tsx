import Link from "next/link";
import { type ComponentProps, forwardRef } from "react";
import { cn } from "~/lib/utils";
import { Spinner } from "./Spinner";

type CommonProps = {
  variant?: "filled" | "outlined";
  color?: "primary" | "secondary" | "tertiary" | "quaternary" | "quinternary" | "senary";
};

const variants = ({ variant = "filled", color = "primary" }: CommonProps) =>
  cn(
    "flex items-center justify-center gap-[0.625rem] rounded-lg px-[1.875rem] py-[0.9375rem] font-bold text-sm leading-1 disabled:bg-white-200 disabled:text-primary-200 disabled:hover:cursor-not-allowed disabled:hover:line-through",
    {
      "bg-primary-100/80 text-white-100 hover:bg-primary-100":
        variant === "filled" && color === "primary",
      "bg-primary-100/10 text-primary-100 hover:bg-primary-100 hover:text-white-100":
        variant === "filled" && color === "secondary",
      "bg-secondary-200 text-white-100 hover:bg-secondary-200/70":
        variant === "filled" && color === "tertiary",
      "bg-secondary-300 text-white-100 hover:bg-secondary-300/70":
        variant === "filled" && color === "quaternary",
      "bg-secondary-100 text-white-100 hover:bg-secondary-100/70":
        variant === "filled" && color === "quinternary",
      "bg-secondary-400 text-white-100 hover:bg-secondary-400/70":
        variant === "filled" && color === "senary",
    },
    {
      "border bg-transparent": variant === "outlined",
      "border-primary-100 text-primary-100 hover:bg-primary-100/70 hover:text-white-100":
        variant === "outlined" && color === "primary",
      "border-secondary-[#ECECF2] text-primary-200 hover:bg-primary-200/70 hover:text-white-100":
        variant === "outlined" && color === "secondary",
      "border-secondary-200 text-secondary-200 hover:bg-secondary-200/70 hover:text-white-100":
        variant === "outlined" && color === "tertiary",
      "border-secondary-300 text-secondary-300 hover:bg-secondary-300/70 hover:text-white-100":
        variant === "outlined" && color === "quaternary",
      "border-secondary-100 text-secondary-100 hover:bg-secondary-100/70 hover:text-white-100":
        variant === "outlined" && color === "quinternary",
      "border-secondary-400 text-secondary-400 hover:bg-secondary-400/70 hover:text-white-100":
        variant === "outlined" && color === "senary",
    }
  );

type LinkProps = CommonProps & React.ComponentProps<typeof Link>;

const LinkButton = forwardRef<HTMLAnchorElement, LinkProps>(
  (
    { className, variant = "filled", color = "primary", children, ...rest },
    ref
  ) => {
    return (
      <Link
        {...rest}
        ref={ref}
        className={cn(variants({ variant, color }), className)}
      >
        {children}
      </Link>
    );
  }
);

LinkButton.displayName = "LinkButton";

type ButtonProps = CommonProps & ComponentProps<"button"> & {
  isLoading?: boolean;
};

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, className, isLoading, ...rest }, ref) => {
    return (
      <button {...rest} ref={ref} className={cn(variants(rest), className)}>
        {isLoading ? <Spinner /> : children}
      </button>
    );
  }
);

Button.displayName = "Button";

export { Button, LinkButton };
