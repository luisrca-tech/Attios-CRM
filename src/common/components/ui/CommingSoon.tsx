import { cn } from '~/lib/utils';

type Props = {
	className?: string;
	message: string;
};

export function CommingSoon({ message, className }: Props) {
	return (
		<div
			className={cn(
				'ml-auto inline-flex items-center rounded-full border bg-white-100 px-2.5 py-0.5 font-bold text-black text-sm transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2',
				className
			)}
		>
			<span>{message}</span>
		</div>
	);
}
