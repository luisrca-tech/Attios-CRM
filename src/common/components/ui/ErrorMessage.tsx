export function ErrorMessage({ message }: { message: string }) {
  return <p className="text-secondary-300 text-sm">{message}</p>;
}
