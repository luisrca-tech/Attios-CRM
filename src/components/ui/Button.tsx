import Link from "next/link";
import { type ComponentProps, forwardRef } from "react";
import { cn } from "~/lib/utils";

type CommonProps = {
  variant?: "filled" | "outlined";
  color?: "primary" | "secondary" | "tertiary" | "quaternary" | "quinternary" | "senary" |  "septenary";
};

const variants = ({ variant = "filled", color = "primary" }: CommonProps) =>
  cn(
    "flex items-center justify-center gap-[0.625rem] rounded-lg px-[1.875rem] py-[0.9375rem] font-bold text-sm leading-1",
    {
      "bg-primary-100/80 text-white-100 hover:bg-primary-100":
        variant === "filled" && color === "primary",
      "bg-primary-100/10 text-primary-100 hover:bg-primary-100 hover:text-white-100":
        variant === "filled" && color === "secondary",
      "bg-white-200 text-primary-200 hover:cursor-not-allowed":
        variant === "filled" && color === "tertiary",
      "bg-secondary-200 text-white-100 hover:bg-secondary-200/70":
        variant === "filled" && color === "quaternary",
      "bg-secondary-300 text-white-100 hover:bg-secondary-300/70":
        variant === "filled" && color === "quinternary",
      "bg-secondary-100 text-white-100 hover:bg-secondary-100/70":
        variant === "filled" && color === "senary",
      "bg-secondary-400 text-white-100 hover:bg-secondary-400/70":
        variant === "filled" && color === "septenary",
    },
    {
      "border bg-transparent": variant === "outlined",
      "border-primary-100 text-primary-100 hover:bg-primary-100/70 hover:text-white-100":
        variant === "outlined" && color === "primary",
      "border-secondary-[#ECECF2] text-primary-200 hover:bg-primary-200/70 hover:text-white-100":
        variant === "outlined" && color === "secondary",
      "border-secondary-200 text-secondary-200 hover:bg-secondary-200/70 hover:text-white-100":
        variant === "outlined" && color === "tertiary",
      "border-secondary-300 text-secondary-300 hover:bg-secondary-300/70 hover:text-white-100":
        variant === "outlined" && color === "quaternary",
      "border-secondary-100 text-secondary-100 hover:bg-secondary-100/70 hover:text-white-100":
        variant === "outlined" && color === "quinternary",
      "border-secondary-400 text-secondary-400 hover:bg-secondary-400/70 hover:text-white-100":
        variant === "outlined" && color === "senary",
    }
  );

type LinkProps = CommonProps & React.ComponentProps<typeof Link>;

const LinkButton = forwardRef<HTMLAnchorElement, LinkProps>(
  (
    { className, variant = "filled", color = "primary", children, ...rest },
    ref
  ) => {
    return (
      <Link
        {...rest}
        ref={ref}
        className={cn(variants({ variant, color }), className)}
      >
        {children}
      </Link>
    );
  }
);

LinkButton.displayName = "LinkButton";

type ButtonProps = CommonProps & ComponentProps<"button">;

const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  ({ children, className, ...rest }, ref) => {
    return (
      <button {...rest} ref={ref} className={cn(variants(rest), className)}>
        {children}
      </button>
    );
  }
);

Button.displayName = "Button";

export { Button, LinkButton };
