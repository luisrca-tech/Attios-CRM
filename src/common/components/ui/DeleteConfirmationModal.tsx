import { useAtom } from "jotai";
import { isOpenConfirmationModal } from "~/common/atoms/is-open-confirmation-modal";
import { Button } from "./Button";
import { Dialog, DialogContent, DialogFooter, DialogHeader, DialogTitle } from "./dialog";

interface DeleteConfirmationModalProps {
  onConfirm: () => void;
  onCancel: () => void;
}

export const DeleteConfirmationModal = ({ onConfirm, onCancel }: DeleteConfirmationModalProps) => {
  const [isOpen, setIsOpen] = useAtom(isOpenConfirmationModal);

  const handleOpenChange = () => {
    setIsOpen(prev => !prev);
  }

  return (
    <Dialog open={isOpen} onOpenChange={handleOpenChange}>
      <DialogContent>
        <DialogHeader>
          <DialogTitle className="text-base font-bold text-black">Are you sure you want to delete this item?</DialogTitle>
        </DialogHeader>
        <DialogFooter>
          <Button type="button" color="quaternary" onClick={onConfirm}>
            Confirm
          </Button>
          <Button type="button" color="secondary" onClick={onCancel}>
            Cancel
          </Button>
        </DialogFooter>
      </DialogContent>
    </Dialog>
  )
}