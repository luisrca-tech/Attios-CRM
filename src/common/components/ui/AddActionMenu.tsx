"use client"

import { useAtom } from "jotai";
import { selectedAddAction } from "~/common/atoms/selected-add-action";
import { addActionItems } from "~/common/constants/addActionItems";
import { Button } from "./Button";
import { CommingSoon } from "./CommingSoon";
import { DropdownMenu, DropdownMenuContent, DropdownMenuItem, DropdownMenuTrigger } from "./dropdown-menu";
import { Icon } from "./Icons/_index";

export function AddActionMenu() { 
  const [selectedModal, setSelectedModal] = useAtom(selectedAddAction);

  const toggleItemModal = (item: typeof addActionItems[number]) => {
    if (item.renderModal) {
      setSelectedModal(item.renderModal());
    } else {
      setSelectedModal(null);
    }
  };

  console.log(selectedModal);

  if (selectedModal) {
    return selectedModal;
  }
  
  return (
      <DropdownMenu>
        <DropdownMenuTrigger asChild className="bg-white-400 data-[state=open]:bg-primary-100">
          <Button className="p-3 hover:bg-white-200/60" color="secondary">
            <Icon.AddButton className="h-4 w-4" />
          </Button>
        </DropdownMenuTrigger>
        <DropdownMenuContent className="bg-white-100 mr-7 mt-2 lg:min-w-[17.8125rem]">
          {addActionItems.map((item) => (
            <DropdownMenuItem 
              onClick={() => toggleItemModal(item)} 
              className={`hover:bg-white-200 ${item.isComingSoon ? "pointer-events-none" : "cursor-pointer"}`}	
              key={item.label}
            >
              <div className="flex items-center justify-center w-9 h-9 bg-primary-100/90 rounded-lg">
                {item.icon}
              </div>
              {item.isComingSoon ? (
                <CommingSoon className="ml-0" message="Coming Soon" />
              ) : (
                <span className="text-black font-bold text-sm leading-6">{item.label}</span>
              )}
            </DropdownMenuItem>
          ))}
        </DropdownMenuContent>
      </DropdownMenu>
  )
}
