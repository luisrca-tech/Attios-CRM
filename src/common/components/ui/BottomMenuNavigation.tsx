'use client';

import { UserButton, useUser } from '@clerk/nextjs';
import Image from 'next/image';
import Link from 'next/link';
import { useState } from 'react';
import {
	Sheet,
	SheetClose,
	SheetContent,
	SheetHeader,
	SheetTitle,
	SheetTrigger
} from '~/common/components/ui/sheet';
import { menuItems } from '~/common/constants/menuItems';
import { cn } from '~/lib/utils';
import { Button } from './Button';
import { CommingSoon } from './CommingSoon';
import { Icon } from './Icons/_index';
import { UserStatusLogged } from './UserStatusLogged';
import { usePathname } from 'next/navigation';
import { AddActionMenu } from './AddActionMenu';

export function BottomMenu() {
	const { user, isLoaded } = useUser();
	const userImage = user?.imageUrl;
	const userName = user?.fullName;
	const [isOpen, setIsOpen] = useState(false);
	const pathname = usePathname();

	const mainItemsMenu = menuItems.slice(0, 4);
	const remainingItemsMenu = menuItems.slice(6);

	const [selectedIcon, setSelectedIcon] = useState<React.ReactNode>(
		<Icon.Menu className="text-primary-200" />
	);

	return (
		<div className="fixed right-0 bottom-0 left-0 z-10 flex h-[11.5lvh] w-full items-center justify-between border-white-200 border-t bg-white-100 px-2 py-[0.875rem] lg:hidden">
			<nav className={cn('flex w-full', isOpen && 'hidden')}>
				<ul className="flex w-full items-center justify-around">
					<li>
						<UserButton
							appearance={{
								elements: {
									avatarBox: 'h-[1,375rem] w-[1,375rem]'
								}
							}}
						/>
					</li>
					{mainItemsMenu.map((item) => (
						<li
							className="relative flex h-12 w-12 items-center justify-center"
							key={item.label}
						>
							<Link
								aria-disabled={item.isComingSoon}
								href={`/${item.label}`}
								className={cn(
									'group flex h-full w-full items-center justify-center',
									item.isComingSoon && 'pointer-events-none opacity-50'
								)}
							>
								<div
									className={cn(
										'-top-[1.125rem] absolute h-[2px] w-full bg-primary-100 transition-transform',
										pathname === `/${item.label}`
											? 'scale-x-100'
											: 'scale-x-0 group-hover:scale-x-100'
									)}
								/>
								<div
									className={cn(
										'relative flex h-full w-full items-center justify-center rounded',
										pathname === `/${item.label}`
											? 'bg-primary-200/10'
											: 'hover:bg-primary-200/10'
									)}
								>
									{item.isComingSoon && (
										<CommingSoon
											message="Soon"
											className="absolute rotate-45 bg-primary-200/10 px-1.5 py-0 text-[0.625rem] text-black"
										/>
									)}
									{item.icon(
										pathname === `/${item.label}` ? '#5E81F4' : '#8181A5'
									)}
								</div>
							</Link>
						</li>
					))}
				</ul>
			</nav>
			<Sheet open={isOpen} onOpenChange={setIsOpen}>
				<SheetTrigger className="group relative flex h-12 w-12 items-center justify-center hover:bg-primary-200/10 hover:outline-none">
					<div className="-top-[1.125rem] absolute h-[2px] w-full scale-x-0 bg-primary-100 transition-transform group-hover:scale-x-100" />
					<div className="flex h-full w-full items-center justify-center rounded hover:bg-primary-200/10">
						{selectedIcon}
					</div>
				</SheetTrigger>
				<SheetContent
					side="bottom"
					className="flex h-full flex-col overflow-y-scroll bg-white-300 pb-3"
				>
					<SheetClose />
					<SheetHeader className="bg-white-100">
						<Button
							variant="outlined"
							color="secondary"
							className="h-10 w-10 border-0 bg-primary-200/10 px-0 hover:bg-primary-200/70"
						>
							<Icon.Search />
						</Button>
						<SheetTitle>Navigation</SheetTitle>
						<AddActionMenu />
					</SheetHeader>
					<div className="mt-[3.625rem] flex flex-col items-center justify-center">
						<div className="mb-[2.3125rem] flex flex-col items-center justify-center">
							<div className="relative">
								<Image
									src={userImage || '/default-avatar.png'}
									alt="user"
									width={94}
									height={94}
									className="rounded-xl"
								/>
								<UserStatusLogged
									userStatus={`${isLoaded ? 'online' : 'offline'}`}
								/>
							</div>
							<span className="font-bold text-2xl text-black leading-[2.375rem]">
								{userName}
							</span>
						</div>
						<nav className="w-full pb-2">
							<ul className="w-full">
								{remainingItemsMenu.map((item) => (
									<li key={item.label} className="w-full px-3 pb-1">
										<SheetClose asChild>
											<Link
												aria-disabled={item.isComingSoon}
												href={`/${item.label}`}
												className={cn(
													'group relative flex items-center gap-7 rounded-[0.625rem] bg-white-100 px-7 py-6 hover:bg-white-200',
													item.isComingSoon && 'pointer-events-none opacity-50'
												)}
												onClick={() =>
													!item.isComingSoon &&
													setSelectedIcon(item.icon('#8181A5'))
												}
											>
												{item.icon('#8181A5')}
												<span className="font-bold text-primary-200/80 text-sm leading-5">
													{item.label}
												</span>
												{item.isComingSoon && (
													<CommingSoon
														message="Coming soon"
														className="absolute right-2"
													/>
												)}
											</Link>
										</SheetClose>
									</li>
								))}
							</ul>
						</nav>
					</div>
				</SheetContent>
			</Sheet>
		</div>
	);
}
