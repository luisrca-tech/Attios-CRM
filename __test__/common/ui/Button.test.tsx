import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Button } from '~/common/components/ui/Button';

describe('Button', () => {
	it('should render', () => {
		const { container } = render(<Button>Click me</Button>);
		expect(container).toMatchSnapshot();
	});
});
