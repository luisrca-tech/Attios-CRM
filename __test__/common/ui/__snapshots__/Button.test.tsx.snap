// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<div>
  <button
    class="flex items-center justify-center gap-[0.625rem] rounded-lg px-[1.875rem] py-[0.9375rem] font-bold text-sm leading-1 disabled:bg-white-200 disabled:text-primary-200 disabled:hover:cursor-not-allowed disabled:hover:line-through bg-primary-100/80 text-white-100 hover:bg-primary-100"
  >
    Click me
  </button>
</div>
`;
